### ISSUE TITLE
I studied the same tutorial

### ISSUE THREAD

**Issue by Dagye95** [LRetroGamer316/Real-tuts-tutorial#1]
What specifically do you need help with, I studied the same tutorial and have had conversations with him maybe i can help.

**Comment by LRetroGamer316** [comment]
I've don't understand how the trail system works, also need help with the menu system.

**Comment by Dagye95** [comment]
i have managed to create the collision system for you, I've also made a few other changes like changing the bar to red when health goes down and the background to grey, I will make more changes but please change up the code so it doesn't seem to similar to my final year proj

**Comment by LRetroGamer316** [comment]
Thank you, I can see that you've added the trail effect to the game and updated the health bar to make it red instead grey. But how come you changed it?

**Comment by Dagye95** [comment]
Ok, so what I've done now, is changed the code in the health bar with the background to white instead of grey and the bar now transitions from green to red when the health is Lower, I find this method to be the best, once a merge it have a look

**Comment by LRetroGamer316** [comment]
Thank you, I'd prefer it that way.

**Comment by LRetroGamer316** [comment]
I downloaded the file, when I ran it i reallly like the changes you made however I accidentally reverted the changes please re-upload them.

**Comment by Dagye95** [comment]
I have done it if you wouldn't mind I'd like to use this conversation in my report

**Comment by LRetroGamer316** [comment]
Yes that is fine, you've helped me a lot with this project and I've come to understand it abit more. So from what I can see you've added a spawn, Game state, trail and menu class. You've managed to cap the fps below 400 when it was reaching the thousands before, that was tricky. I see you also tried to add the pause feature but that is quite working properly but it's good enough. 

I hope that helps with your report :)

**Comment by Dagye95** [comment]
import java.awt.Color;
import java.awt.Graphics;
import java.awt.Rectangle;

public class SmartEnemy extends Gameobject{

	private Handler handler;
	private Gameobject player;
	
	public SmartEnemy(int x, int y, ID id, Handler handler) {
		super(x, y, id);
		this.handler = handler;
		
		for(int i = 0; i < handler.object.size(); i++){
			if(handler.object.get(i).getId() == ID.Player) player = handler.object.get(i);
		}
		
	}

	public Rectangle getBounds(){
		return new Rectangle((int)x, (int)y, 16, 16);
	}
	
	public void tick() {
		x += velX;
		y += velY;
		
		float diffX = x - player.getX() - 8;
		float diffY = y - player.getY() - 8; 
		float distance = (float) Math.sqrt(((x-player.getX())*(x-player.getX()) + (y-player.getY())*(y-player.getY())));
	
		velX = (float) ((-1.0/distance) * diffX);
		velY = (float) ((-1.0/distance) * diffY);

		if(y <= 0 || y >= Game.HEIGHT - 32) velY *= -1;
		if(x <= 0 || x >= Game.WIDTH - 16) velX *= -1;
		
		handler.addObject(new Trail((int)x, (int)y, ID.Trail, Color.green, 16, 16, 0.02f, handler));
		
	}

	
	public void render(Graphics g) {

		g.setColor(Color.GREEN);
		g.fillRect((int)x, (int)y, 16, 16);
	}

}

**Comment by Dagye95** [comment]
That is going to help me a lot thank you, here is a code for the smart enemy

**Comment by Dagye95** [comment]
this enemy will follow players ;)
